### DISCUSSION TITLE
Digital output to arduino is not observable on arduino

### DISCUSSION THREAD

**Discussion by ameerlawen** [bonsai-rx/bonsai#1214] [Q&A]
Hi, 

I've been using bonsai to detect behavioral changes in mice, along with recording a video. The arduino is created and DigitalOutput module shows the wanted output, yet electrically the arduino does not represent the binary voltage changes (I connected an LED to make sure this works - the LED does not flicker although the digital output alternates between 0 and 1 on bonsai monitor).  

Has anyone dealt with arduino DigitalOutput problems similar to what I described? 

* I changed multiple arduinos
* The arduino is recognized and connected properly

Best, 
Amir
<img width="1242" alt="bonsaiQ" src="https://user-images.githubusercontent.com/73176387/210118808-c47e365e-b8a9-46e0-8fb4-1681e2dc2a64.PNG">

**Comment by bruno-f-cruz** [comment]
Hi.
Try to move the `CreateArduino` operator to the top of the workflow and try again. It might be an initialization problem (check #1167 for an example).

Hope it helps.
Bruno

**Reply by ameerlawen** [reply]
Hi, 

Unfortunately it didn't help. Also changing the portName as in #1167 seems to not help/adds identification problems in another laptop.

It is important to know that it worked for me last month.

The arduino is connected properly (pin13 flickers when I start the workflow) and there are no errors, but simply although I see the wanted output in DigitalOutput module, it is not translated to an electrical pulse in arduino.

also after simplifying it (doing exercise 3 in closed-loop systems tutorial), nothing changes. 

also attaching a picture of the arduino, the ON light is on and the Rx light is on (most of the time in a noisy way), as long as the workflow is played.

![20230102_220127](https://user-images.githubusercontent.com/73176387/210295357-68ecc2a3-2770-4b89-a760-1960207d5134.jpg)

![bonsaiQ3](https://user-images.githubusercontent.com/73176387/210295456-b0bd7948-53aa-44e4-91ef-7f87dc001eea.PNG)

![bonsaiQ3-2](https://user-images.githubusercontent.com/73176387/210295469-ae9e977d-fb22-4e83-a038-c57a263e3371.PNG)

I will be happy if you have any more insights.

best,
Amir

**Reply by bruno-f-cruz** [reply]
Just to make sure, is Firmata flashed in the Arduino Board?

**Reply by ameerlawen** [reply]
Hi, 

I have flashed Firmata perviously but when I began to face this problem I changed the arduino and did not flash Firmata. Today I uploaded Firmata to the new arduino, yet my workflow did not work right away. Afterwards I tried to run the closed loop tutorial exercise, it worked. I copied the arduino initialization modules to my workflow, then my workflow worked although the modules were identical.

I'm not sure exactly what was causing the problem initially. But it works now. Thank you very much. 

Best, 
Amir
